ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -275,33 +268,103 @@ trait EventStorageBackend {
+  )(connection: Connection): Vector[EventStorageBackend.Entry[Raw.FlatEvent]]
+
+  def fetchTreeCreateEvents(
+      eventSequentialIds: Iterable[Long],
+      allFilterParties: Set[Ref.Party],
+  )(connection: Connection): Vector[EventStorageBackend.Entry[Raw.TreeEvent]]
+
+  def fetchTreeNonConsumingEvents(
+      eventSequentialIds: Iterable[Long],
+      allFilterParties: Set[Ref.Party],
+  )(connection: Connection): Vector[EventStorageBackend.Entry[Raw.TreeEvent]]
+
+  def fetchIds_create_stakeholders(
+      partyFilter: Party,
+      templateIdFilter: Option[Identifier],
+      startExclusive: Long,
+      endInclusive: Long,
+      limit: Int,
+  )(connection: Connection): Vector[Long]
+
+  def fetchIds_create_nonStakeholderInformees(
+      partyFilter: Party,
+      startExclusive: Long,
+      endInclusive: Long,
+      limit: Int,
+  )(connection: Connection): Vector[Long]
+
+  /** @return at most `limit` event sequential ids from the given range
+    *         that match the specified party and template filters,
+    *         and which are retrieved from the stakeholders filter table
+    *         for consuming events.
+    *         (A party is present in the stakeholders filter table for consuming events
+    *         if and only if there is a consuming exercise on the corresponding contract
+    *         on which this party is a stakeholder.)
+    */
+  def fetchIds_consuming_stakeholders(
       partyFilter: Party,
       templateIdFilter: Option[Identifier],
       startExclusive: Long,
       endInclusive: Long,
       limit: Int,
   )(connection: Connection): Vector[Long]
+
+  def fetchIds_consuming_nonStakeholderInformees(
+      partyFilter: Party,
+      startExclusive: Long,
+      endInclusive: Long,
+      limit: Int,
+  )(connection: Connection): Vector[Long]
+
+  def fetchIds_nonConsuming_informees(
+      partyFilter: Party,
+      startExclusive: Long,
+      endInclusive: Long,
+      limit: Int,
+  )(connection: Connection): Vector[Long]
+
   def activeContractEventBatch(
       eventSequentialIds: Iterable[Long],
       allFilterParties: Set[Party],
       endInclusive: Long,
   )(connection: Connection): Vector[EventStorageBackend.Entry[Raw.FlatEvent]]
-  def flatTransaction(
-      transactionId: TransactionId,
-      filterParams: FilterParams,
+
+  def fetchIdsFromTransactionMeta(
+      transactionId: Ref.TransactionId
+  )(connection: Connection): Option[(Long, Long)]
+
+  def fetchFlatTransaction(
+      firstEventSequentialId: Long,
+      lastEventSequentialId: Long,
+      requestingParties: Set[Party],
   )(connection: Connection): Vector[EventStorageBackend.Entry[Raw.FlatEvent]]
-  def transactionTreeEvents(
-      rangeParams: RangeParams,
-      filterParams: FilterParams,
-  )(connection: Connection): Vector[EventStorageBackend.Entry[Raw.TreeEvent]]
-  def transactionTree(
-      transactionId: TransactionId,
-      filterParams: FilterParams,
+
+  def fetchTreeTransaction(
```

Comment:
as pointed out in offline discussion and in the task itself, for the pointwise tree and flat transaction retrieval I would strongly suggest to explore an alternative approach involving reusing as much as possible from the flat and stream population functions. Ideally so that StorageBackend layer is not even impacted.